src/database/core/view/EventRegistration.ts

```diff
@@ -11,269 +13,250 @@ import { Path } from "../util/Path";
+    const ctx = this.context_;
+    if (eventData.getEventType() === 'cancel') {
+      assert(this.cancelCallback_, 'Raising a cancel event on a listener with no cancel callback');
+      const cancelCB = this.cancelCallback_;
+      return function () {
+        // We know that error exists, we checked above that this is a cancel event
+        cancelCB.call(ctx, eventData.error);
+      };
+    } else {
+      const cb = this.callbacks_[eventData.eventType];
+      return function () {
+        cb.call(ctx, eventData.snapshot, eventData.prevName);
+      }
     }
   }
-};
 
-
-/**
- * @inheritDoc
- */
-ChildEventRegistration.prototype.matches = function(other) {
-  if (other instanceof ChildEventRegistration) {
-    if (!this.callbacks_ || !other.callbacks_) {
-      return true;
-    } else if (this.context_ === other.context_) {
-      var otherCount = getCount(other.callbacks_);
-      var thisCount = getCount(this.callbacks_);
-      if (otherCount === thisCount) {
-        // If count is 1, do an exact match on eventType, if either is defined but null, it's a match.
-        //  If event types don't match, not a match
-        // If count is not 1, exact match across all
-
-        if (otherCount === 1) {
-          var otherKey = /** @type {!string} */ (getAnyKey(other.callbacks_));
-          var thisKey = /** @type {!string} */ (getAnyKey(this.callbacks_));
-          return (thisKey === otherKey && (
-              !other.callbacks_[otherKey] ||
-              !this.callbacks_[thisKey] ||
-              other.callbacks_[otherKey] === this.callbacks_[thisKey]
+  /**
+   * @inheritDoc
+   */
+  matches(other) {
+    if (other instanceof ChildEventRegistration) {
+      if (!this.callbacks_ || !other.callbacks_) {
+        return true;
+      } else if (this.context_ === other.context_) {
+        const otherCount = getCount(other.callbacks_);
+        const thisCount = getCount(this.callbacks_);
+        if (otherCount === thisCount) {
+          // If count is 1, do an exact match on eventType, if either is defined but null, it's a match.
+          //  If event types don't match, not a match
+          // If count is not 1, exact match across all
+
+          if (otherCount === 1) {
+            const otherKey = /** @type {!string} */ (getAnyKey(other.callbacks_));
+            const thisKey = /** @type {!string} */ (getAnyKey(this.callbacks_));
+            return (thisKey === otherKey && (
+                !other.callbacks_[otherKey] ||
+                !this.callbacks_[thisKey] ||
+                other.callbacks_[otherKey] === this.callbacks_[thisKey]
               )
-          );
-        } else {
-          // Exact match on each key.
-          return this.callbacks_.every(function(cb, eventType) {
-            return other.callbacks_[eventType] === cb;
-          });
+            );
+          } else {
+            // Exact match on each key.
+            return this.callbacks_.every(<any>function (cb, eventType) {
```

Comment:
In case it is indeed a bug, we could add this function to `src/utils/obj.ts`:
```js
export const every = function(obj, fn) {
  for (let key in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      if (!fn(key, obj[key])) {
        return false;
      }
    }
  }
  return true;
};
```

and then replace this block of code with:
```ts
return every(this.callbacks_, (eventType, cb) => other.callbacks_[eventType] === cb);
```

Let me know! :)